### ISSUE TITLE
Arrow syntax with multiline params

### ISSUE THREAD

**Issue by samdj** [gandm/language-babel#149]
```js
const fn = (
  arg, {
    a: arg2
  }
) => arg2 + arg

const fn2 = ({
    a: arg2
 }) => arg2 + 2
```

The arrow doesn't highlight in these examples (Gets treated as relational operators).

**Comment by gandm** [comment]
Expected as per bullet point 3. https://github.com/gandm/language-babel/issues/46

**Comment by gandm** [comment]
I've edited the bullet point and title in the referenced issue to make this more clear. It is a limitation of grammar in general that one can't backtrack to re-annotate previous code. The `=>` is incorrect but so are the scopes of the previous `(` and `)`. These should be shown as a function. By the time the `=>` occurs I can't go back and re-parse the function arguments. 

 Maybe I can correctly scope the` =>` though. I'll consider that and see if it has any unwanted consequences.

**Comment by gandm** [comment]
@samdj Just released 2.17.0 that scopes arrows. Still doesn't parse it as a correct function though.

**Comment by samdj** [comment]
Looks like good compromise @gandm,
